src/LargeXlsx/Util.cs

```diff
@@ -28,107 +28,112 @@ OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN
-            var columnName = CachedColumnNames[columnIndex - 1];
-            if (columnName == null)
-            {
-                columnName = GetColumnNameInternal(columnIndex);
-                CachedColumnNames[columnIndex - 1] = columnName;
-            }
-            return columnName;
-        }
-
-        private static string GetColumnNameInternal(int columnIndex)
-        {
-            var columnName = new StringBuilder(3); // This has been measured to be faster than string concatenation
-            while (true)
-            {
-                if (columnIndex > 26)
-                {
-                    columnIndex = Math.DivRem(columnIndex - 1, 26, out var rem);
-                    columnName.Insert(0, (char)('A' + rem));
-                }
-                else
-                {
-                    columnName.Insert(0, (char)('A' + columnIndex - 1));
-                    return columnName.ToString();
-                }
-            }
-        }
-
-        public static double DateToDouble(DateTime date)
-        {
-            var days = date.Subtract(ExcelEpoch).TotalDays + 1;
-            // Excel wrongly assumes that 1900 is a leap year:
-            // https://docs.microsoft.com/en-us/office/troubleshoot/excel/wrongly-assumes-1900-is-leap-year
-            if (date >= Date19000301) days++;
-            return days;
-        }
-
-        public static int BoolToInt(bool value) => value ? 1 : 0;
-
-        public static string EnumToAttributeValue<T>(T enumValue)
-        {
-            var s = enumValue.ToString();
-            return char.ToLowerInvariant(s[0]) + s.Substring(1);
-        }
-
-        // Hashing procedure courtesy of https://docs.microsoft.com/en-us/openspecs/office_file_formats/ms-offcrypto/1357ea58-646e-4483-92ef-95d718079d6f
-        public static byte[] ComputePasswordHash(string password, byte[] saltValue, int spinCount)
-        {
-            var hasher = new SHA512Managed();
-            var hash = hasher.ComputeHash(saltValue.Concat(Encoding.Unicode.GetBytes(password)).ToArray());
-            for (var i = 0; i < spinCount; i++)
-            {
-                var iterator = BitConverter.GetBytes(i);
-                if (!BitConverter.IsLittleEndian)
-                    Array.Reverse(iterator);
-                hash = hasher.ComputeHash(hash.Concat(iterator).ToArray());
-            }
-            return hash;
-        }
-    }
+	internal static class Util
+	{
+		private const int MaxColumnCount = 16384;
+		private static readonly DateTime ExcelEpoch = new DateTime(1900, 1, 1);
+		private static readonly DateTime Date19000301 = new DateTime(1900, 3, 1);
+		private static readonly string[] CachedColumnNames = new string[MaxColumnCount];
+
+		public static string EscapeXmlText(string value)
+		{
+			return System.Security.SecurityElement.Escape(RemoveInvalidXmlChars(value));
```

Comment:
In my understanding this is stricter than needed, because, if I recall correctly, apostrophe and double quotes are not required to be escaped in XML text, just on XML attributes. Did you find any particular reason to use SecurityElement.Escape here, too?

Reply:
Not actually. I hoped that it would solve my problem regarding illegal XML characters, but it did not help in some cases and i needed to make the second function anyway for escaping these as well.